### PR TITLE
Breaking: Remove `ecmaFeatures` from `eslint:recommended`

### PR DESCRIPTION
**Pull request by alberto**
**What is the purpose of this pull request? (put an "X" next to item)**

[ ] Documentation update
[ ] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/rule-proposal.md))
[ ] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/rule-change-proposal.md))
[ ] Add autofixing to a rule
[ ] Add a CLI option
[ ] Add something to the core
[X] Other, please explain:

**What changes did you make? (Give an overview)**
Remove `ecmaFeatures` from `eslint:recommended`. It's not a top level property since eslint v2.0.0.

**Is there anything you'd like reviewers to focus on?**
Should be tagged: 'Chore'? Not sure, since it's not really internal.
Even if it changes `eslint:recommended` I don't think it should be considered a breaking change.

### CODE REVIEW

**Comment by not-an-aardvark** [comment]
Are there any scenarios where this change could cause errors? For example, could this cause a rule to crash if it's expecting `ecmaFeatures` to be present in the config?

**Comment by alberto** [comment]
I guess it could cause errors if it was assuming the existence of the property, not doing something like:

```
const ecmaFeatures = context.ecmaFeatures || {};

if ecmaFeatures.globalReturn) {...};
```

But then it would be checking against the wrong property, so the rule wouldn't be working as intended.

**Comment by platinumazure** [comment]
I'd feel better if we just did this with 4.0 release "just in case" (IMO, it is making a potentially breaking alteration on a public API and we'd be honoring the semver contract). But I agree that the change "should" be harmless.

**Comment by platinumazure** [comment]
Added "breaking" label for now, per this excerpt from the [maintainer guide](http://eslint.org/docs/maintainer-guide/issues.html#when-an-issue-is-opened):

> Breaking Changes: if you **suspect** or can verify that a change would be breaking, label it as “Breaking”.

Emphasis mine. I'd like it discussed and if consensus is that the change is non-breaking, awesome, we can remove the label and merge as a chore.

**Comment by platinumazure** [comment]
Just to move this along (since I helped get it stuck in the first place), I'm going to add this to TSC agenda.

**TSC Summary**: The `eslint:recommended` should no longer export `ecmaFeatures` in its `parserOptions`, since that has been deprecated by espree long ago. The purpose of this PR is to remove `ecmaFeatures`. However, a concern was raised that any custom or plugin rules may rely on `context.ecmaFeatures` which could trip people up who are using `eslint:recommended` without being aware of this. I (platinumazure) labeled this as a potentially breaking change, with the idea that if the team disagrees, the label can be removed and this could be labeled as a chore.

**TSC Question**: Should this change be merged only with a major release, or can we safely merge it as a chore?

**Comment by not-an-aardvark** [comment]
This was accepted by the TSC as a breaking change in today's TSC meeting.

File: tests/lib/config.js [hunk 1]
```diff
@@ -358,9 +358,6 @@ describe("Config", () => {
                         browser: true,
                         node: false
                     },
-                    ecmaFeatures: {
-                        globalReturn: false
```
Comment (platinumazure) [inline tests/lib/config.js]: globalReturn is still a valid ecmaFeature, isn't it? (If we just don't need it, though, that's a different story.)

Comment (not-an-aardvark) [inline tests/lib/config.js]: I think `ecmaFeatures` is valid as part of the `parserOptions` object, but it's not valid as a top-level property.

Comment (platinumazure) [inline tests/lib/config.js]: Right you are, I wasn't looking carefully enough.

**Pull request merged**